runtime/j9vm/asgct.cpp

```diff
@@ -120,6 +120,11 @@ protectedASGCT(J9PortLibrary *portLib, void *arg)
 		walkState.walkThread = currentThread;
 		walkState.skipCount = 0;
 		walkState.maxFrames = parms->depth;
+		/* JIT cache may allocate, which we don't want inside a signal handler,
+		 * Flag is cleared in the caller (so both the normal and exception cases
+		 * clear it).
+		 */
+		currentThread->jitArtifactSearchCache = (void*)((UDATA)currentThread->jitArtifactSearchCache | J9_STACKWALK_NO_JIT_CACHE);
```

Comment:
Can an allocation occur further up in `J9JITExceptionTable *metaData = jitConfig->jitGetExceptionTableFromPC(currentThread, rip);` as well?

Reply:
No, the backend `jit_artifact_search` is strictly a read operation.

Reply:
The stack walk is not allowed to fail due to memory allocation failure, so all of the allocations have a fallback - global buffer for stack/local map, and not using the metadata cache if it fails to allocate. We don't walk slots here, so the map case never occurs.